src/index.ts

```diff
@@ -51,75 +46,58 @@ class BaseClient {
       grpc.credentials.createInsecure()
     );
 
-    this.opts = opts;
-    // Use optional chaining for cleaner code
-    this.owner = opts.owner ?? undefined;
+    // this.opts = opts;
+    // this.owner = opts.owner ?? undefined;
```

Comment:
Mainly two things are done:

1. De-coupled the owner and client options from the constructor of Client, as they can be passed in to auth functions from the application level later. Rpc endpoint is the only hard requirement for constructing the client.

2. De-coupled private key from this file, since I don’t think our SDK requires the dependency of `ethers`. What we really need is the signed message. However, in test cases, `ethers` in included for convenience.